app/controllers/heroku/resources_controller.rb

```diff
@@ -1,30 +1,26 @@
 module Heroku
   class ResourcesController < ApplicationController
     def create
-      heroku_uuid = params[:uuid]
-      oauth_grant_code = params[:oauth_grant][:code]
-      plan = params[:plan]
+      sandwich = create_sandwich
+      enqueue_token_exchange_job(sandwich)
 
-      Sandwich.create!(
-        heroku_uuid: heroku_uuid,
-        oauth_grant_code: oauth_grant_code,
-        plan: plan,
-      )
-
-      ExchangeGrantTokenJob.perform_later(
-        heroku_uuid: heroku_uuid,
-        oauth_grant_code: oauth_grant_code,
-      )
+      if plan == Sandwich::BASE_PLAN # synchronous provisioning
+        message = 'Thanks for using Sudo Sandwich.'
+        status = 200
+      else # async provisioning
```

Comment:
And down here you might set state to `provisioning`, and then let your background job (or whatever) that toggles to `provisioned` update the sandwich instance state there.

I know this is a little artificial, it's not a feature we need, but if we want this to demonstrate things to partners being a little artificial is OK.

Reply:
good idea, added `state` and updates at a179225